rust/protocol/src/curve/curve25519.rs

```diff
@@ -133,16 +191,24 @@ impl KeyPair {
         bool::from(cap_r_check.as_bytes().ct_eq(&cap_r))
     }
 
+    /// Return a public key that another party can use to validate against [Self::verify_signature].
     pub fn public_key(&self) -> &[u8; PUBLIC_KEY_LENGTH] {
         &self.public_key
     }
 
+    /// Return a private key that can be converted into a public key with [derive_public_key].
     pub fn private_key(&self) -> &[u8; PRIVATE_KEY_LENGTH] {
         &self.private_key
     }
 }
 
-pub fn derive_public_key(private_key: &[u8; 32]) -> [u8; 32] {
+/// Produce the result of [KeyPair::public_key], when given the output of [KeyPair::private_key].
+///```
+/// # use libsignal_protocol::curve_unstable::curve25519::*;
+/// let kp = KeyPair::new(&mut rand::thread_rng());
+/// assert!(kp.public_key() == &derive_public_key(kp.private_key()));
+///```
+pub fn derive_public_key(private_key: &[u8; PRIVATE_KEY_LENGTH]) -> [u8; PUBLIC_KEY_LENGTH] {
```

Comment:
"Produce the public key corresponding to the given private key". KeyPair is not the fundamental API here, even if "a private/public key pair" is the underlying abstraction.